libpod/boltdb_state_internal.go

```diff
@@ -154,6 +162,17 @@ func (s *BoltState) getDBCon() (*bolt.DB, error) {
 	return db, nil
 }
 
+// Close a connection to the database.
+// MUST be used in place of `db.Close()` to ensure proper unlocking of the
+// state.
+func (s *BoltState) closeDBCon(db *bolt.DB) error {
+	err := db.Close()
+
+	s.dbLock.Unlock()
```

Comment:
Should we unlock if err != nil?

Reply:
I think it's safe to do it unconditional - we won't be using the connection anymore, even if it failed to close, so overriding the file locks won't be a big deal.